src/server/api/cycleData/table/clearTable.ts

```diff
@@ -0,0 +1,72 @@
+import { Response } from 'express'
+
+import { CycleDataRequest } from '@meta/api/request'
+
+import { AssessmentController } from '@server/controller/assessment'
+import { CycleDataController } from '@server/controller/cycleData'
+import { MetadataController } from '@server/controller/metadata'
+import Requests from '@server/utils/requests'
+
+export const clearTable = async (req: CycleDataRequest, res: Response) => {
+  try {
+    const { countryIso, assessmentName, cycleName, sectionName, tableName } = req.query
+
+    const { assessment, cycle } = await AssessmentController.getOneWithCycle({
+      assessmentName,
+      cycleName,
+      metaCache: true,
+    })
+
+    const tableSpec = await MetadataController.getTable({
+      tableName,
+      cycle,
+      assessment,
+    })
+
+    const rows = await MetadataController.getRows({
+      assessment,
+      tableName,
+      includeCols: true,
+    })
+
+    const nodes = rows.reduce((acc, row) => {
+      if (!row.props.variableName || row.props.readonly) {
+        return acc
+      }
+      const { variableName } = row.props
+
+      const columnNames = tableSpec.props.columnNames[cycle.uuid].reduce((acc, colName) => {
+        const col = row.cols.find((c) => c.props.colName === colName)
+        if (!col.props.colName || col.props.readonly) {
+          return acc
+        }
+        return [...acc, col.props.colName]
+      }, [])
+
+      return [
+        ...acc,
+        ...columnNames.map((colName) => ({
+          tableName,
+          variableName,
+          colName,
+          value: { raw: null },
+        })),
+      ]
+    }, [])
+
+    await CycleDataController.persistNodeValues({
+      nodeUpdates: {
+        assessment,
+        cycle,
+        countryIso,
+        nodes,
+      },
+      sectionName,
+      user: Requests.getUser(req),
+    })
+
```

Comment:
wouldn't it be more efficient one singe delete query ?

Reply:
so from api you call ```CycleDataController.clearTable({...})```
in controller something:
```
cont nodes = NodeRepository.deleteTableNodes({...})
const nodeUpdates = {assessment, cycle.... , nodes}
activity log...
scheduleUpdateDependencies()
```

Reply:
we have this, but it had strange behavior so I disabled
I will investigate

https://github.com/openforis/fra-platform/pull/2460/files#diff-1b8eb1ba24c47332a99b498ef87aeade956fcd8bc3e46ceebbdc9fa5290c3888